nabs/preprocessors.py

```diff
@@ -112,9 +112,29 @@ class State:
         daq_has_triggered = False
 
     def daq_first_cycle(msg):
-        yield from bps.configure(daq, events=events, duration=duration,
-                                 record=record, use_l3t=use_l3t,
-                                 controls=list(motor_cache))
+        if events is not None:
+            yield from bps.configure(
+                daq,
+                events=events,
+                record=record,
+                use_l3t=use_l3t,
+                controls=list(motor_cache),
+            )
+        elif duration is not None:
+            yield from bps.configure(
+                daq,
+                duration=duration,
+                record=record,
+                use_l3t=use_l3t,
+                controls=list(motor_cache),
+            )
+        else:
+            yield from bps.configure(
+                daq,
+                record=record,
```

Comment:
This assumes the daq was configured first, but what if it wasn't?  (the answer is a RuntimeError, as before 👍 )

Reply:
That's right, the user will get the previously seen error but it will actually be correct this time!